app/models/user.rb

```diff
@@ -3,6 +3,8 @@ class User < ApplicationRecord
   ADMIN_ROLE = 'admin'.freeze
   MODERATOR_ROLE = 'moderator'.freeze
   ROLES = [SALE_ROLE, ADMIN_ROLE, MODERATOR_ROLE].freeze
+  VALID_PHONE_REGEX = /\A[+]?[\d\-.() ]+\z/
+  VALID_EMAIL_REGEX = /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i
```

Comment:
try https://github.com/validates-email-format-of/validates_email_format_of for email validation.